src/test/java/eu/europeana/metis/sandbox/service/workflow/HarvestServiceImplTest.java

```diff
@@ -0,0 +1,82 @@
+package eu.europeana.metis.sandbox.service.workflow;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import eu.europeana.metis.sandbox.common.exception.ServiceException;
+import eu.europeana.metis.utils.ZipFileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.mock.web.MockMultipartFile;
+
+
+@ExtendWith(MockitoExtension.class)
+public class HarvestServiceImplTest {
+
+  private static final ZipFileReader zipfilereader = new ZipFileReader();
+
+  private static final HarvestServiceImpl harvestService = new HarvestServiceImpl();
+
+  @Test
+  void harvestServiceFromURL_expectSuccess() throws IOException {
+
+    Path dataSetPath = Paths.get("src", "test", "resources", "zip", "dataset-valid.zip");
+    assertTrue(Files.exists(dataSetPath));
+
+    var expectedRecords = zipfilereader.getContentFromZipFile(Files.newInputStream(dataSetPath));
+
+    var records = harvestService.harvest(dataSetPath.toUri().toString());
+
+    // In spite records being the same size, they are not in the same order.
+    // So comparing with assertArrayEquals() will output a failed assertion
+    // assertArrayEquals(expectedRecords.toArray(),records.toArray());
+    assertEquals(expectedRecords.size(), records.size());
+  }
+
+  @Test
+  void harvestServiceFromUploadFile_expectSuccess() throws IOException {
+
+    Path dataSetPath = Paths.get("src", "test", "resources", "zip", "dataset-valid.zip");
+
+    assertTrue(Files.exists(dataSetPath));
+
+    MockMultipartFile dataset = new MockMultipartFile("dataset", "dataset.txt", "text/plain",
+        Files.newInputStream(dataSetPath));
+
+    var expectedRecords = zipfilereader.getContentFromZipFile(Files.newInputStream(dataSetPath));
+
+    var records = harvestService.harvest(dataset);
+    // In spite records being the same size, they are not in the same order.
+    // So comparing with assertArrayEquals() will output a failed assertion
+    // assertArrayEquals(expectedRecords.toArray(),records.toArray());
+    assertEquals(expectedRecords.size(), records.size());
+  }
+
+  @Test
+  void harvestService_expectFailNonExistingFile() {
```

Comment:
This seems to be a check for the url not the file as described in the name of the method.
Also it actually fails with indeed a `ServiceException` but I think it fails because the String passed to it is not in a URL format and that it's a missing file.
A check we need to have, but the way it's handled is misleading.